### PR TITLE
DDF-3338 upgrade maven-dependency-plugin to 3.0.2

### PR DESCRIPTION
**Pull request by oconnormi**
#### What does this PR do?

Upgrades maven-dependency-plugin to 3.0.2

#### Who is reviewing it? 

@mcalcote @LinkMJB 

#### Select relevant component teams: 

@codice/build 

#### Choose 2 committers to review/merge the PR. 

@lessarderic 
@shaundmorris 

#### What are the relevant tickets?
[DDF-3338](https://codice.atlassian.net/browse/DDF-3338)

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by oconnormi** [comment]
build now

**Comment by oconnormi** [comment]
build now

**Comment by oconnormi** [comment]
Seems like this version might hurt more than help, already got two build failures with issues resolving/downloading dependencies

**Comment by LinkMJB** [comment]
@oconnormi it looks like those dependency resolution failures were legitimate, it was the same dependency that failed in both builds. The flag that led me looking deeper into it, is that BFA didn't identify it as a Nexus issue, meaning that the artifact was actually missing rather than it timed-out resolving. AFAICT, it looks like the dependency is available now (Nexus negative cache invalidation timing?)

**Comment by oconnormi** [comment]
@LinkMJB well in that case we can try again 😄 
Build Now

**Comment by oconnormi** [comment]
@LinkMJB looks like its still having problems finding certain dependencies. I'll have to look into it more

**Comment by LinkMJB** [comment]
@oconnormi I think that's the same dependency it was failing to find before 😢

**Comment by oconnormi** [comment]
build now

**Comment by oconnormi** [comment]
I'm starting to think there is a bug in this version of the maven-dependency-plugin. It continuously fails to download the same artifact each time the build is run. The error message it provides is `Unable to find/resolve artifact. Could not find artifact ddf.platform.solr:platform-solr-server-standalone:jar:2.12.0-SNAPSHOT` when building the `catalog-core-solr` module. The pom for that module actually depends on a `test-jar` not a `jar` artifact from the `platform-solr-server-standalone` module. The `platform-solr-server-standalone` module declares a main artifact of type `war` and its build does not produce a regular jar or attach one anywhere to the deploy. The only artifacts created by this modules build are the war, some xml files, and the test-jar. I think this version of the dependency plugin is either incorrectly looking for a jar instead of a test-jar, or it is just expecting that a jar (which is the default packaging type) exists for each dependency even if it is not the requested artifact.

**Comment by oconnormi** [comment]
I can confirm there is a bug in dependency plugin 3.0.2. It does is not correctly translating the type `test-jar` to the real values of `type:jar`, `classifier:tests` the way it used to. instead it is just getting the jar artifact. This would result in always getting the incorrect artifact when requesting test-jars, since it would always grab the production jar instead.

**Comment by oconnormi** [comment]
Created a bug for this: https://issues.apache.org/jira/browse/MDEP-587, but I'm thinking we won't be upgrading to this version 😄

**Comment by oconnormi** [comment]
Added a workaround for the bug I linked above

Build Now

File: catalog/core/catalog-core-federationstrategy/pom.xml [hunk 1]
```diff
@@ -48,7 +48,10 @@
         <dependency>
             <groupId>ddf.catalog.core</groupId>
             <artifactId>catalog-core-api-impl</artifactId>
-            <type>test-jar</type>
+            <type>jar</type>
+            <!-- Bug in dependency-plugin 3.0.2 requires using classifier for test-jars instead of type.
```
Comment (shaundmorris) [inline catalog/core/catalog-core-federationstrategy/pom.xml]: thanks for adding the comments in here

Comment (oconnormi) [reply catalog/core/catalog-core-federationstrategy/pom.xml]: its the type of thing that I myself would have changed back if I didn't see a comment like that 😄

**Comment by oconnormi** [comment]
@shaundmorris I found them all by running `ag '<type>test-jar'` from the root of the project

File: catalog/core/catalog-core-federationstrategy/pom.xml (hunk 1 above)
Comment (shaundmorris) [reply catalog/core/catalog-core-federationstrategy/pom.xml]: yes, I would have forgotten by then

**Comment by oconnormi** [comment]
not sure why those tests failed, trying again.

Build Now

**Pull request merged**